Remotion/Data/DomainObjects.UnitTests/Queries/Configuration/QueryConfigurationTest.cs

```diff
@@ -126,98 +128,77 @@ public void GetDefaultQueryFilePath_BaseDirectory ()
 #endif
     public void GetDefaultQueryFilePath_WithRelativeSearchPath ()
     {
-      AppDomainSetup setup = AppDomain.CurrentDomain.SetupInformation;
-      setup.ApplicationBase = Path.GetPathRoot (AppContext.BaseDirectory);
-      setup.DynamicBase = Path.GetTempPath ();
-      setup.PrivateBinPath = Path.GetFullPath (AppContext.BaseDirectory).Substring (setup.ApplicationBase.Length); // make a relative path
+      IAppContextProvider provider = MockRepository.GenerateStub<IAppContextProvider>();
+      provider.Stub (_ => _.BaseDirectory).Return (Path.GetPathRoot (AppContext.BaseDirectory));
```

Comment:
Copy `TestContext.CurrentContext.TestDirectory\queries.xml` into a previously created temporary folder and use this temp folder as `BaseDirectory`.

Creating directory, copy file and removing directory afterwards may be done only once in fixture setup and teardown.